torchdata/datapipes/iter/util/common.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+
+from typing import Callable, Iterable, Iterator, TypeVar
+
+from torch.utils.data.datapipes.utils.common import _check_unpickable_fn
+from torchdata.datapipes import functional_datapipe
+
+from torchdata.datapipes.iter import IterDataPipe, Mapper
+
+T_co = TypeVar("T_co", covariant=True)
+
+
+class MapTemplateIterDataPipe(Mapper[T_co]):
+    def __init__(self, source_datapipe, input_col=None, output_col=None) -> None:
+        fn = self._map
+        super().__init__(source_datapipe, fn=fn, input_col=input_col, output_col=output_col)
+
+    def _map(self, *args, **kwargs) -> T_co:
+        raise NotImplementedError
```

Comment:
BTW, we might want to use `abc. abstractmethod` here to guarantee to implement the method at the construct time.